### PR TITLE
add webirc kiwi client

### PR DESCRIPTION
**Pull request by benharri**

### CODE REVIEW

**Comment by daurnimator** [comment]
I think it may be better to include this as part of the ircd application?

Lets run it on irc.hashbang.sh instead of a new domain?

**Comment by benharri** [comment]
not a bad thought. The trick is to set up nginx to forward something like /webirc to the ircd's websocket port (which could be changed to internal), which I'm not sure how to do inside the existing ircd application

**Comment by daurnimator** [comment]
> The trick is to set up nginx to forward something like /webirc to the ircd's websocket port (which could be changed to internal),

That's done with an ingress specification.

File: ircd/files/ircd.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ server:
 
         # Native websockets over TLS on a nonstandard (non-443) port. This is typically not
         # used directly, since application firewalls may block it:
-        ":443":
+        ":8097":
```
Comment (daurnimator) [inline ircd/files/ircd.yaml:40]: no need to change this

File: ircd/resources.yaml [hunk 2]
```diff
@@ -39,6 +39,11 @@ spec:
           readOnly: true
         - mountPath: /ircd/db
           name: irc-db
+      - name: webchat
+        image: hashbang/webirc:latest
```
Comment (daurnimator) [inline ircd/resources.yaml:43]: drop off the `:latest`

File: ircd/resources.yaml [hunk 3]
```diff
@@ -65,13 +70,49 @@ spec:
   - name: irc
     port: 6697
     targetPort: 6697
-  - name: websockets
```
Comment (daurnimator) [inline ircd/resources.yaml:68]: I think you need to point port 443 at the ingress?

Comment (daurnimator) [reply ircd/resources.yaml:68]: also port 80 I guess?

File: ircd/resources.yaml [hunk 4]
```diff
@@ -39,6 +39,11 @@ spec:
           readOnly: true
         - mountPath: /ircd/db
           name: irc-db
+      - name: webchat
+        image: hashbang/webirc:latest
+        ports:
+        - containerPort: 80
+          name: http
```
Comment (daurnimator) [inline ircd/resources.yaml:46]: it's https not http right?

File: ircd/resources.yaml (hunk 3 above)
Comment (benharri) [reply ircd/resources.yaml:68]: does that belong in the loadbalancer or the ingress?

Comment (daurnimator) [reply ircd/resources.yaml:68]: normally the ingress creates a load balancer *for* us; however for our case here we want to reuse an existing one

File: ircd/resources.yaml [hunk 5]
```diff
@@ -39,6 +39,11 @@ spec:
           readOnly: true
         - mountPath: /ircd/db
           name: irc-db
+      - name: webchat
+        image: hashbang/webirc
+        ports:
+        - containerPort: 80
```
Comment (daurnimator) [inline ircd/resources.yaml:45]: Shouldn't this be `443`?

File: ircd/resources.yaml [hunk 6]
```diff
@@ -65,13 +70,55 @@ spec:
...
+    secretName: webirc-tls
+  rules:
+  - host: irc.hashbang.sh
+    http:
+      paths:
+      - path: /*
+        backend:
+          serviceName: webirc
+          servicePort: 80
+      - path: /webirc
```
Comment (daurnimator) [inline ircd/resources.yaml:116]: funny that /webirc points at not webirc :P
Perhaps `/websocket` or something like that would be better?

File: ircd/resources.yaml [hunk 7]
```diff
@@ -65,13 +70,55 @@ spec:
...
+  selector:
+    app: webirc
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: webirc-ingress
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    cert-manager.io/cluster-issuer: "letsencrypt-prod"
```
Comment (daurnimator) [inline ircd/resources.yaml:101]: Don't want this annotation; we already have the `irc.hashbang.sh` cert in the `irc-certs` secret.

File: ircd/resources.yaml [hunk 8]
```diff
@@ -65,13 +70,55 @@ spec:
...
+    external-dns.alpha.kubernetes.io/hostname: irc.hashbang.sh
```
Comment (daurnimator) [inline ircd/resources.yaml:102]: remove this; it points at the LB not the ingress

**Pull request merged**